src/util/market_prices.ts

```diff
@@ -32,6 +38,35 @@ export const getZeroXPriceInUSD = async (): Promise<BigNumber> => {
     return zeroXPriceInUSD;
 };
 
+export const getZeroXPriceInWeth = async (): Promise<BigNumber> => {
+    /* Gets all the available sell orders of 0x and takes the lowest price, is there are no 0x sells orders available, the price is calculated using coin market */
+    const web3Wrapper = await getWeb3Wrapper();
+    let zeroXPrice = new BigNumber('0');
+    if (web3Wrapper) {
+        const networkId = await web3Wrapper.getNetworkIdAsync();
+        const knownTokens = getKnownTokens(networkId);
+        const zeroXToken = knownTokens.getTokenBySymbol('ZRX');
+        let sellOrders = await getAllOrdersAsUIOrders(zeroXToken);
+        if (sellOrders && sellOrders.length > 0) {
+            sellOrders = sellOrders
+                .filter(order => order.side === OrderSide.Sell)
+                .sort((o1, o2) => o1.price.comparedTo(o2.price));
+            /* Check if there is any order that could fill the needs of 0x */
+            const orderWithEnoughZeroX = sellOrders.find(sellOrder => {
+                return sellOrder.size.cmp(MAKER_FEE) === 1;
+            });
+            zeroXPrice = orderWithEnoughZeroX ? orderWithEnoughZeroX.price : zeroXPrice;
+        }
+        if (!zeroXPrice) {
+            const zeroXInUSD = await getZeroXPriceInUSD();
+            const ethInUSD = await getEthereumPriceInUSD();
+            zeroXPrice = zeroXInUSD.mul(ethInUSD);
```

Comment:
Q: How does this work? It seems like it's multiplying a price in USD by a price in USD, does that make sense?

Reply:
Thanks for checking this, I'd made a mistake, it's a division what is needed to calculate that, now it's changed